src/main/java/uk/gov/hmcts/divorce/caseworker/event/CaseworkerUpdateContactDetails.java

```diff
@@ -0,0 +1,181 @@
+package uk.gov.hmcts.divorce.caseworker.event;
+
+import org.springframework.stereotype.Component;
+import uk.gov.hmcts.ccd.sdk.api.CCDConfig;
+import uk.gov.hmcts.ccd.sdk.api.ConfigBuilder;
+import uk.gov.hmcts.ccd.sdk.api.Event;
+import uk.gov.hmcts.ccd.sdk.api.FieldCollection;
+import uk.gov.hmcts.ccd.sdk.type.Organisation;
+import uk.gov.hmcts.ccd.sdk.type.OrganisationPolicy;
+import uk.gov.hmcts.divorce.ccd.PageBuilder;
+import uk.gov.hmcts.divorce.common.model.Applicant;
+import uk.gov.hmcts.divorce.common.model.Application;
+import uk.gov.hmcts.divorce.common.model.CaseData;
+import uk.gov.hmcts.divorce.common.model.MarriageDetails;
+import uk.gov.hmcts.divorce.common.model.Solicitor;
+import uk.gov.hmcts.divorce.common.model.State;
+import uk.gov.hmcts.divorce.common.model.UserRole;
+
+import static uk.gov.hmcts.divorce.common.model.UserRole.CASEWORKER_COURTADMIN_CTSC;
+import static uk.gov.hmcts.divorce.common.model.UserRole.CASEWORKER_COURTADMIN_RDU;
+import static uk.gov.hmcts.divorce.common.model.UserRole.CASEWORKER_LEGAL_ADVISOR;
+import static uk.gov.hmcts.divorce.common.model.UserRole.CASEWORKER_SUPERUSER;
+import static uk.gov.hmcts.divorce.common.model.UserRole.SOLICITOR;
+import static uk.gov.hmcts.divorce.common.model.access.Permissions.CREATE_READ_UPDATE;
+import static uk.gov.hmcts.divorce.common.model.access.Permissions.READ;
+
+@Component
+public class CaseworkerUpdateContactDetails implements CCDConfig<CaseData, State, UserRole> {
+    public static final String CASEWORKER_UPDATE_CONTACT_DETAILS = "caseworker-update-contact-details";
+
+    @Override
+    public void configure(final ConfigBuilder<CaseData, State, UserRole> configBuilder) {
+        PageBuilder pageBuilder = new PageBuilder(configBuilder
+            .event(CASEWORKER_UPDATE_CONTACT_DETAILS)
+            .forAllStates()
+            .name("Update contact details")
+            .description("Update contact details")
+            .explicitGrants()
+            .grant(CREATE_READ_UPDATE,
+                CASEWORKER_COURTADMIN_CTSC,
+                CASEWORKER_COURTADMIN_RDU)
+            .grant(READ,
+                SOLICITOR,
+                CASEWORKER_SUPERUSER,
+                CASEWORKER_LEGAL_ADVISOR));
+
+        pageDetails(pageBuilder);
```

Comment:
page fields should go in a separate class if you check other events

Reply:
Created a new page class